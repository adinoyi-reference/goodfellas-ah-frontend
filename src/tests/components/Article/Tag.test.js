import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import AddTags, { Tags } from '../../../components/article/Tags';
import Root from '../../../root';

const mockAddTags = mount(
  <Root>
    <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
      <AddTags />
    </MemoryRouter>
  </Root>,
);

const publishedArticleError = {
  status: {
    success: true,
    error: false,
  },
};

const testAddTagsMount = shallow(
  <Tags publishedArticle={['home']} publishedArticleError={publishedArticleError} />,
);

describe('Tags UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(mockAddTags);
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Tags Functionality', () => {
  describe('handle functionality', () => {
    it('should simulate publish', () => {
      const preventDefault = jest.fn();
      const publishArticleButton = testAddTagsMount.find('.publish-article-button');
      publishArticleButton.simulate('click', {
        preventDefault,
      });
    });

    it('should simulate submit', () => {
      const mockedEvent = {
        preventDefault: jest.fn(),
        target: {
          reset: jest.fn(),
        },
      };
      const submitButton = testAddTagsMount.find('form');

      const instance = testAddTagsMount.instance();
      instance.refs = {
        getTags: {
          disabled: false,
          value: 'true',
        },
      };

      submitButton.simulate('submit', mockedEvent);
    });

    it('should simulate submit but fail to publish', () => {
      const mockedEvent = {
        preventDefault: jest.fn(),
        target: {
          reset: jest.fn(),
        },
      };
      const submitButton = testAddTagsMount.find('form');

      const instance = testAddTagsMount.instance();
      instance.refs = {
        getTags: {
          disabled: false,
          value: 'true',
        },
      };
      instance.props.publishedArticleError.status.error = true;

      submitButton.simulate('submit', mockedEvent);
    });

    it('should simulate click', () => {
      const mockedEvent = {
        preventDefault: jest.fn(),
        target: {
          previousSibling: {
            textContent: 'home',
          },
          reset: jest.fn(),
        },
      };

      const clickButton = testAddTagsMount.find('.new-tag-cancel').at(0);

      const instance = testAddTagsMount.instance();

      instance.state.articleStatus = true;
      instance.state.tags = ['true'];
      instance.refs = {
        getTags: {
          disabled: false,
          value: '',
        },
      };

      clickButton.simulate('click', mockedEvent);

      instance.state.tags = ['true', 'safe', 'wish', 'think', 'home'];

      clickButton.simulate('click', mockedEvent);
    });
  });
});
