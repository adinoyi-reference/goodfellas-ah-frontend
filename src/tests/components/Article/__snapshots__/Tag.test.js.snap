// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags UI render features component should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testkey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testkey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testkey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Tags)>
          <Tags
            addTags={[Function]}
            publishArticle={[Function]}
            publishedArticleError={
              Object {
                "publishedArticle": Object {},
                "status": Object {
                  "error": false,
                  "success": false,
                },
              }
            }
            tags={Array []}
            tagsError={Array []}
          >
            <div
              className="tags"
            >
              <div
                className="tags-main"
              >
                <h2>
                  Add tags to your article
                </h2>
                <div
                  className="tags-main-details"
                >
                  <p>
                    Help your target audience find this article by adding up to 5 tags
                  </p>
                  <form
                    onSubmit={[Function]}
                  >
                    <input
                      placeholder="Add a tag..."
                      type="text"
                    />
                  </form>
                  <div
                    className="tags-wrapper"
                  />
                  <div
                    className="publish-article-button"
                    onClick={[Function]}
                  >
                    <Button
                      className="btn hero-section-greenbutton"
                      title="Publish"
                      type="submit"
                    >
                      <button
                        className="btn hero-section-greenbutton"
                        type="submit"
                      >
                        Publish
                      </button>
                    </Button>
                  </div>
                </div>
              </div>
            </div>
          </Tags>
        </Connect(Tags)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
